src/lib/StorageFileApi.ts

```diff
@@ -288,25 +288,30 @@ export class StorageFileApi {
   }
 
   /**
-   * Retrieve URLs for assets in public buckets
+   * Retrieve URLs for assets in public buckets and encapsulates it in a return object
    *
    * @param path The file path to be downloaded, including the path and file name. For example `folder/image.png`.
    */
   getPublicUrl(
     path: string
   ): {
-    data: { publicURL: string } | null
-    error: Error | null
-    publicURL: string | null
+    data: { publicURL: string }
+    error: null
+    publicURL: string
   } {
-    try {
-      const _path = this._getFinalPath(path)
-      const publicURL = `${this.url}/object/public/${_path}`
-      const data = { publicURL }
-      return { data, error: null, publicURL }
-    } catch (error) {
-      return { data: null, error, publicURL: null }
-    }
+    const _path = this._getFinalPath(path)
+    const publicURL = `${this.url}/object/public/${_path}`
+    const data = { publicURL }
+    return { data, error: null, publicURL }
```

Comment:
I think you're right in saying this is solved mainly in #60. While it is cleaner to do without the try/catch, with the new custom error type added in #60, it is nice for proper type inference and code consistency.

Reply:
On second thought I am torn on this one. It would be very smooth to just do:
```
const { publicURL } = storage.from('my-bucket').getPublicUrl('profile.png')
```

Reply:
I don't think simply adding a dummy `error` field to a function to ensure code consistency is really a good idea. That error will never be anything but `null`, at least based on my understanding and my testing.

It's not all about ease of use here, but about confusion and introducing bloat in the code base of users. Granted, I am making this seem more of an issue than it actually is. An extra `if` statement in the code will not degrade performance in any meaningful way but i just don't see the reason behind having the error there since it will never, at least in the current code base, reach an error.

Reply:
Yep, I get where you're coming from. If not consistency then just for not introducing breaking changes. The new TS type will also show it always being null anyway and we can just mark it as deprecated.